src/main/java/io/netty/incubator/codec/http3/Http3ControlStreamInboundHandler.java

```diff
@@ -15,22 +15,46 @@
  */
 package io.netty.incubator.codec.http3;
 
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.Channel;
 import io.netty.channel.ChannelHandler;
 import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
 import io.netty.channel.socket.ChannelInputShutdownEvent;
+import io.netty.incubator.codec.quic.QuicChannel;
+import io.netty.incubator.codec.quic.QuicStreamChannel;
+import io.netty.incubator.codec.quic.QuicStreamType;
 import io.netty.util.ReferenceCountUtil;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.GenericFutureListener;
+import io.netty.util.concurrent.Promise;
+
+import java.nio.channels.ClosedChannelException;
+
+import static io.netty.incubator.codec.http3.Http3CodecUtils.connectionError;
+import static io.netty.incubator.codec.http3.Http3ErrorCode.QPACK_ENCODER_STREAM_ERROR;
+import static io.netty.incubator.codec.http3.Http3SettingsFrame.HTTP3_SETTINGS_QPACK_BLOCKED_STREAMS;
+import static io.netty.incubator.codec.http3.Http3SettingsFrame.HTTP3_SETTINGS_QPACK_MAX_TABLE_CAPACITY;
+import static io.netty.util.internal.ThrowableUtil.unknownStackTrace;
 
 final class Http3ControlStreamInboundHandler extends Http3FrameTypeValidationHandler<Http3ControlStreamFrame> {
+    private static final GenericFutureListener<Future<? super QuicStreamChannel>> CLOSE_ON_FAILURE = f -> {
+        if (!f.isSuccess()) {
+            ((QuicStreamChannel) f.getNow()).close();
```

Comment:
huh... if `!f.isSuccess()` `getNow()` will return null as `cause` will be filled.